app/components/listing/RelativeOpeningTime.jsx

```diff
@@ -1,62 +1,94 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import moment from 'moment';
+
+// TODO Until we use actual timestamps, this one constant converts everything to PST below
+const timezoneOffset = 8;
 
 export class RelativeOpeningTime extends React.Component {
   static parseAsDate(fourOrThreeDigitNumber) {
     const fourDigitNumber = `0000000${fourOrThreeDigitNumber}`.slice(-4);
-    const hour = Number(`${fourDigitNumber}`.slice(0, 2)) * 1000 * 60 * 60;
+    const hour = (Number(`${fourDigitNumber}`.slice(0, 2)) + timezoneOffset) * 1000 * 60 * 60;
     const mins = Number(`${fourDigitNumber}`.slice(2, 4)) * 1000 * 60;
     // console.log(fourOrThreeDigitNumber, '->', fourDigitNumber, '->', [hour, mins]);
     return new Date(hour + mins);
   }
 
-  static parseSchedule(schedule_days, currentDate = new Date()) {
+  // Turns a schedule day into an exact date/timestamp for the next relevant day from now
+  static convertScheduleDayToOpenCloseTimestamps(schedule_day, currentDate = moment()) {
+    const today = currentDate.clone().day();
+    const daysOfWeek = RelativeOpeningTime.daysOfWeek.slice();
+    const todayAndAfter = daysOfWeek.splice(today);
+    const orderedDaysOfTheWeek = todayAndAfter.concat(daysOfWeek);
+
+    const daysFromToday = orderedDaysOfTheWeek.indexOf(schedule_day.day);
+    const openingTime = RelativeOpeningTime.parseAsDate(schedule_day.opens_at);
+    const closingTime = RelativeOpeningTime.parseAsDate(schedule_day.closes_at);
+
+    const opens_at = currentDate
+      .clone()
+      .add(daysFromToday, 'days')
+      .startOf('day')
+      .add(openingTime.getHours(), 'hours')
+      .add(openingTime.getMinutes(), 'minutes');
+
+    const closes_at = currentDate
+      .clone()
+      .add(daysFromToday, 'days') // If the close is before the open, add a day
```

Comment:
I understand why we need to do the logic described by the comment, but I don't actually see where we are adding the extra day. To me, it looks like `opens_at` and `closes_at` will both have the same day, even if the `closes_at` hour is before the `opens_at` hour.

Reply:
Hmm you're right. I think I accidentally removed this in cleaning up my own code... Adding it back.